packages/react/src/ui-react/components/form/parameter-type.tsx

```diff
@@ -157,33 +167,98 @@ export function StringParamTextField<
     );
 }
 
-export function SubscriptionIdParamDropdown<
+export function StorageAccountDropdown<
     V extends FormValues,
     K extends Extract<keyof V, string>
 >(props: ParamControlProps<V, K>): JSX.Element {
     const { param } = props;
     const value = param.value == null ? undefined : String(param.value);
+
+    const [loading, setLoading] = useState<boolean>(true);
+    const [storageAccounts, setStorageAccounts] = useState<StorageAccount[]>(
+        []
+    );
     const id = useUniqueId("form-control", props.id);
+    const service: StorageAccountService = inject(
+        DependencyName.StorageAccountService
+    );
+    const form = param.parentForm;
+    const [subscriptionId, setSubscriptionId] = useState<string>(
+        form.values.subscriptionId as string
+    );
+    const [validationStatus, setValidationStatus] =
+        useState<ValidationStatus | null>();
 
-    const [loading, setLoading] = React.useState<boolean>(true);
-    const [subscriptions, setSubscriptions] = React.useState<
+    useAsyncEffect(async () => {
+        let accounts: StorageAccount[] = [];
+        try {
+            if (subscriptionId) {
+                accounts = await service.list(subscriptionId);
+            }
+            setValidationStatus(null);
+        } catch (error) {
+            setValidationStatus(new ValidationStatus("error", error + ""));
+        } finally {
+            setStorageAccounts(accounts);
+            setLoading(false);
+        }
+    }, [subscriptionId]);
+
+    useEffect(() => {
+        const handler = form.on("change", (values: FormValues) =>
```

Comment:
I don't think an on change event listener is necessary since the most recent form value is always passed down in the props.